pkg/discovery/util/pod_util.go

```diff
@@ -276,3 +281,127 @@ func GetPodGrandInfo(kclient *client.Clientset, pod *api.Pod) (string, string, e
 
 	return kind, name, nil
 }
+
+// WaitForPodReady checks the readiness of a given pod with a retry limit and a timeout, whichever
+// comes first. If a nodeName is provided, also checks that the hosting node matches that in the
+// pod specification
+//
+// TODO:
+// Use k8s watch API to eliminate the need for polling and improve efficiency
+func WaitForPodReady(client *client.Clientset, namespace, podName, nodeName string,
+	retry int, interval time.Duration) error {
+	// check pod readiness with retries
+	timeout := time.Duration(retry+1) * interval
+	err := goutil.RetrySimple(retry, timeout, interval, func() (bool, error) {
+		return checkPodNode(client, namespace, podName, nodeName)
+	})
+	// log a list of unique events that belong to the pod
+	// warning events are logged in Error level, other events are logged in Info level
+	podEvents := getPodEvents(client, namespace, podName)
```

Comment:
Why is the *api.Pod not passed in with checkPodNode() and getPodEvents(), so the two calls to the api client to get the pod can be avoided?

Reply:
For ``checkPodNode()``, the api will need to be called each time to get the updated pod status.
For ``getPodEvents()``, the api is called once.

I think in the future, if we ever run into performance issue, we probably want to use k8s watch api to avoid polling.